Yoast/javascript
# Yoastseo/2196 rewrite analyze to use structured tree

## Summary

This PR can be summarized in the following changelog entry:

* _[not-user-facing]_ Rewrites the SEO and readability analysis such that it uses both the original assessors as well the new tree assessors. The results of both are combined and their scores aggregated.

## Relevant technical choices:

* The `wrapTryCatchAroundAction` that handles the errors thrown in the web worker is now partly asynchronous, to make it work with the new asynchronous nature of the analysis.

## Test instructions

This PR can be tested by following these steps:

* Start the example.
* Type in a (HTML) text.
* Check if the analysis works as expected, and does show an error in the console.
    * A few errors are known already: #38 and #61.
* Run the tests, check if they pass.

Fixes https://github.com/Yoast/YoastSEO.js/issues/2196
Fixes https://github.com/Yoast/YoastSEO.js/issues/2205
